include/aspect/particle/property/crystal_preferred_orientation.h

```diff
@@ -222,6 +240,39 @@ namespace aspect
           unsigned int
           get_number_of_minerals();
 
+          /**
+           * @brief Deterimens the deformation type from the deformation type selector.
+           *
+           * @param stress
+           * @param water_content
+           * @return DeformationType
+           */
+          DeformationType
+          determine_deformation_type(const DeformationTypeSelector deformation_type_selector,
+                                     const Point<dim> &position,
+                                     const double temperature,
+                                     const double pressure,
+                                     const Tensor<1,dim> &velocity,
+                                     const std::vector<double> &compositions,
+                                     const SymmetricTensor<2,dim> &strain_rate,
+                                     const SymmetricTensor<2,dim> &compressible_strain_rate,
+                                     const double water_content) const;
+
+          /**
+           * @brief Computes the deformation type given the stress and water content according to the
+           * table in Karato 2008.
+           *
+           * @param stress
+           * @param water_content
+           */
+          DeformationType
+          determine_deformation_type_karato_2008(const double stress,
+                                                 const double water_content) const;
+
+          std::array<double,4>
+          reference_resolved_shear_stress_from_deformation_type(DeformationType deformation_type,
```

Comment:
maybe document ``maxvalue``? units?

Reply:
It returns the relative strength of the slip planes, which for olivine are 4. This is (more or less) how it is done in the fortran code. I have been thinking to generalize it (some minerals have way more slip planes), but I would rather to do that in a follow up pull request, because this is probably already complicated enough as it is. I will add better documentation.